server/src/main/java/org/apache/cassandra/sidecar/db/schema/SidecarSchema.java

```diff
@@ -181,4 +186,23 @@ protected void reportSidecarSchemaInitialized()
     {
         vertx.eventBus().publish(ON_SIDECAR_SCHEMA_INITIALIZED.address(), "SidecarSchema initialized");
     }
+
+    /**
+     * Returns true when the schema should be created by this Sidecar instance. This currently
+     * depends on whether the election of single instance executor process is available, and
+     * the schemas are of type {@link InitializeOnSingleInstanceExecutor}, and the local Sidecar
+     * is a single instance executor. For all other types of schemas or if the election of leader
+     * process is unavailable we initialize the schemas.
+     *
+     * @param schema the schema to test
+     * @return {@code true} if the schema should be created by this Sidecar instance, {@code false} otherwise
+     */
+    protected boolean shouldCreateSchema(@Nullable AbstractSchema schema)
+    {
+        if (singleInstanceExecutor != null && schema instanceof InitializeOnSingleInstanceExecutor)
+        {
+            return singleInstanceExecutor.isLocalSidecarSingleInstanceExecutor();
+        }
+        return true;
+    }
```

Comment:
I'm not sure if `singleInstanceExecutor` can actually be null (it doesn't look that way)... could we drop the null check here? If so, this method simplifies a bit:
```suggestion
    protected boolean shouldCreateSchema(@Nullable AbstractSchema schema)
    {
        return singleInstanceExecutor.isLocalSidecarSingleInstanceExecutor() || 
               !(schema instanceof InitializeOnSingleInstanceExecutor);
    }
```

Reply:
you don't really need to inject an implantation, so I made it generic so you can omit this feature